### PR TITLE
Feat: Implement the function in the reader contract

### PR DESCRIPTION
**Pull request by tevrat-aksoy**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature
- Testing

## What is the current behavior?

- implemented reader library
- Tests not completed because of the missing libraries

Resolves: #428

### CODE REVIEW

File: src/tests/reader/test_reader.cairo [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+    start_prank(role_store.contract_address, caller_address);
+    role_store.grant_role(caller_address, role::MARKET_KEEPER);
+    stop_prank(role_store.contract_address);
+
+    let salt = poseidon_hash_span(array!['SATORU_MARKET', 0, 0, 0, 0].span());
+
+    // Test logic
+
+    data_store.set_market(key, salt, market);
+    let market_by_key = reader::get_market(data_store, key);
```
Comment (zarboq) [inline src/tests/reader/test_reader.cairo:70]: Shouldn't you use get_market_by_salt in this test ?

**Review by zarboq — CHANGES_REQUESTED** [review]
Nice PR :rocket: 

Left 1 comment on test. Also make reader a deployable contract, not just a library

File: src/tests/reader/test_reader.cairo (hunk 1 above)
Comment (tevrat-aksoy) [reply src/tests/reader/test_reader.cairo:70]: yes, you are right  I will fix it

**Review by zarboq — APPROVED** [review]
Good job :rocket:

**Pull request merged**